src/tiledb/cloud/_results/codecs.py

```diff
@@ -113,6 +116,25 @@ def decode(cls, data: bytes) -> object:
         return json.loads(data)
 
 
+class ArrowDataFrameCodec(Codec["pandas.DataFrame"]):
+    """Encodes Pandas DataFrames as Arrow data."""
+
+    NAME = "arrow_dataframe"
+    # We don't actually serve this MIME type; it is subject to change.
+    MIME = "application/vnd.tiledb.arrow-dataframe"
+
+    @classmethod
+    def encode(cls, data: "pandas.DataFrame") -> bytes:
+        """Converts a Pandas array to the byte format of an Arrow table."""
+        tbl = pyarrow.Table.from_pandas(data)
+        return ArrowCodec.encode(tbl)
```

Comment:
IMO it would be worth catching `ArrowInvalid` here and providing a contextual rethrow, because a user may have no idea why such an error could occur when dealing with a pandas object.

```
import tiledb, numpy as np, pandas as pd, tiledb.cloud
import pyarrow as pa

class W:
    def __init__(self, v):
        self.o = v

df = pd.DataFrame({"foo": pd.Series([W("bar")])})
df
pa.Table.from_pandas(df)
```
->
```
ArrowInvalid: ('Could not convert <__main__.W object at 0x1157cc160> with type W: did not recognize Python value type when inferring an Arrow data type', 'Conversion failed for column foo with type object')
```